biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/summary/WriteSummary.scala

```diff
@@ -159,15 +166,21 @@ class WriteSummary(val parent: SummaryQScript) extends InProcessFunction with Co
         db.createOrUpdateSetting(qscript.summaryRunId, pipelineId, None, None, None, ConfigUtils.mapToJson(q.summarySettings).nospaces)
     }
 
-    (for (f <- qscript.functions.par) yield f match {
-      case f: BiopetJavaCommandLineFunction with Version =>
-        Some(db.createOrUpdateExecutable(qscript.summaryRunId, f.configNamespace, f.getVersion, f.getJavaVersion,
-          javaMd5 = BiopetCommandLineFunction.executableMd5Cache.get(f.executable), jarPath = Option(f.jarFile).map(_.getAbsolutePath)))
-      case f: BiopetCommandLineFunction with Version =>
-        Some(db.createOrUpdateExecutable(qscript.summaryRunId, f.configNamespace, f.getVersion, Option(f.executable)))
-      case f: Configurable with Version =>
-        Some(db.createOrUpdateExecutable(qscript.summaryRunId, f.configNamespace, f.getVersion))
-      case _ => None
+    val pipeFunctions = (for (f <- qscript.functions) yield f match {
+      case f: BiopetCommandLineFunction => f.pipesJobs
+      case _                            => Nil
+    }).flatten
+    (for (f <- qscript.functions ++ pipeFunctions) yield {
```

Comment:
You are already looping through all `qscript.functions` above. Perhaps there's a a way of doing it all in one loop.